src/Microsoft.AspNetCore.Razor/CodeGenerators/CSharpCodeWriter.cs

```diff
@@ -319,6 +319,20 @@ public CSharpCodeWriter WriteMethodInvocation(string methodName, bool endLine, p
                 .WriteEndMethodInvocation(endLine);
         }
 
+        public CSharpCodeWriter WriteAutoPropertyDeclaration(string accessibility, string typeName, string name)
```

Comment:
I assume the view component as tag helper feature needs easy auto-property generation. Correct?

But why does this method needs to exist in Razor? MVC or any external code could implement this as an extension method:
``` c#
public static CSharpCodeWriter WriteAutoPropertyDeclaration(
    this CSharpCodeWriter writer,
    string accessibility,
    string typeName,
    string name)
{
    return writer
        .Write(accessibility)
        // ...
        .WriteLine();
}
```